File: packages/shared/src/types/product.ts
export interface CategoryDto {
    id: string;
    name: string;
}

export interface ProductVariantDto {
    id: string;
    size: string;
    color: string;
    stockQuantity: number;
}

export interface ProductDto {
    id: string;
    name: string;
    description: string;
    imagePath: string;
    imagePaths: string[];
    price: number;
    category: CategoryDto;
    variants: ProductVariantDto[];
}

export interface CategoryCreateRequest {
    name: string;
}

export interface CategoryUpdateRequest {
    name: string;
}

export interface ProductVariantCreateRequest {
    size: string;
    color: string;
    stockQuantity: number;
}

export interface ProductVariantUpdateRequest {
    size: string;
    color: string;
    stockQuantity: number;
}

export interface ProductCreateRequest {
    name: string;
    description: string;
    price: number;
    categoryId: string;
    imagePath?: string;
    imagePaths?: string[];
    variants: ProductVariantCreateRequest[];
}

export interface ProductUpdateRequest {
    name: string;
    description: string;
    price: number;
    categoryId: string;
    imagePath?: string;
    imagePaths?: string[];
}

File: backend/PADIDI.API/Models/Product.cs
namespace PADIDI.API.Models;

public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string ImagePath { get; set; } = string.Empty;
    public List<string> ImagePaths { get; set; } = new();
    public decimal Price { get; set; }
    public Guid CategoryId { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;

    public Category Category { get; set; } = null!;
    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
}
